content_api/publish/service.py

```diff
@@ -38,10 +38,20 @@ def publish(self, item, subscribers=[]):
 
         :param item: item to publish
         """
-
         if not self._filter_item(item):
-            doc = self.formatter._transform_to_ninjs(item, self.subscriber)
             now = utcnow()
+            try:
+                # do we already have an item id db?
+                original_item = self.find_one(req=None, _id=item.get('guid') or item['_id'])
+            except KeyError:
+                original_item = None
+                check_original = True
+            else:
+                # if yes, we need to use previous firstpublished data
+                item.setdefault('firstpublished',
```

Comment:
I'm still not sure why is this needed, it should be set on item when publishing and corrections shouldn't update it anyhow, so what's the usecase here?

Reply:
imo it should be set when user clicks on publish, before it enters the output handling, and not when it's pushed to contentapi. so if you get an item from content api or via ftp or whatever there will be the same timestamp.

Reply:
OK I see, I'll move it.